crates/bevy_reflect/src/serde/type_data.rs

```diff
@@ -42,3 +53,67 @@ impl SerializationData {
         self.ignored_field_indices.is_empty()
     }
 }
+
+/// Contains data relevant to the automatic reflect powered serialization of an enum.
+#[derive(Debug, Clone)]
+pub struct EnumSerializationData {
+    variant_serialization_data: HashMap<Cow<'static, str>, HashSet<usize>>,
+}
+
+impl EnumSerializationData {
+    /// Creates a new `EnumSerializationData` instance given:
+    ///
+    /// - `ignored_variants`: the iterator of member variant-indices pairs to be ignored during serialization.
+    /// Indices are assigned only to reflected members, those which are not reflected are skipped.
+    pub fn new<TName, TFields, TVariant>(ignored_variants: TVariant) -> Self
+    where
+        TName: Into<Cow<'static, str>>,
+        TFields: Iterator<Item = usize>,
+        TVariant: Iterator<Item = (TName, TFields)>,
+    {
```

Comment:
This function signature :sweat_smile: . I see what you are doing here, but could it be possible to do the following instead of having a `TFields` parameter?

```
        TVariant: Iterator<Item = (TName, &[usize])>,
```

Reply:
Hm, I'm having difficulties making it work. It's giving me errors saying it's missing type information.

I'll just leave it as is unless you or someone else has a better way of handling it haha